apps/backend/src/modules/users/user.repository.ts

```diff
@@ -3,60 +3,175 @@ import { type UserPasswordDetails } from "~/modules/users/libs/types/types.js";
-		const users = await this.userModel.query().execute();
+		const users = await this.userModel
+			.query()
+			.select([
+				"users.id",
+				"firstName",
+				"lastName",
+				"users.group_id as groupId",
+				"email",
+				"groups.id",
+				"groups.key as groupKey",
+				"groups.name as groupName",
+			])
+			.innerJoin("groups", "users.group_id", "groups.id")
+			.leftJoin(
+				"groups_to_permissions",
+				"groups.id",
+				"groups_to_permissions.group_id",
+			)
+			.leftJoin(
+				"permissions",
+				"groups_to_permissions.permission_id",
+				"permissions.id",
+			)
+			.execute();
 
-		return users.map((user) => UserEntity.initialize(user));
+		return users.map((user) =>
+			UserEntity.initialize({
+				email: user.email,
+				firstName: user.firstName,
+				group: user.group ? GroupEntity.initialize(user.group) : null,
+				groupId: user.groupId,
+				id: user.id,
+				lastName: user.lastName,
+				passwordHash: user.passwordHash,
+				passwordSalt: user.passwordSalt,
+			}),
+		);
 	}
 
 	public async findByEmail(email: string): Promise<null | UserEntity> {
-		const user = await this.userModel.query().where("email", email).first();
+		const user = await this.userModel
+			.query()
+			.where("email", email)
+			.withGraphJoined("group.permissions")
+			.first();
 
-		return user ? UserEntity.initialize(user) : null;
+		if (!user) {
+			return null;
+		}
+
+		return UserEntity.initialize({
+			email: user.email,
+			firstName: user.firstName,
+			group: user.group ? GroupEntity.initialize(user.group) : null,
+			groupId: user.groupId,
+			id: user.id,
+			lastName: user.lastName,
+			passwordHash: user.passwordHash,
+			passwordSalt: user.passwordSalt,
+		});
 	}
 
 	public async findById(id: number): Promise<null | UserEntity> {
-		const user = await this.userModel.query().findById(id);
+		const user = await this.userModel
+			.query()
+			.findById(id)
+			.withGraphJoined("group.permissions")
+			.first();
 
-		return user ? UserEntity.initialize(user) : null;
+		if (!user) {
+			return null;
+		}
+
+		return UserEntity.initialize({
+			email: user.email,
+			firstName: user.firstName,
+			group: user.group ? GroupEntity.initialize(user.group) : null,
+			groupId: user.groupId,
+			id: user.id,
+			lastName: user.lastName,
+			passwordHash: user.passwordHash,
+			passwordSalt: user.passwordSalt,
+		});
 	}
 
 	public async findPasswordDetails(
 		email: string,
 	): Promise<null | UserPasswordDetails> {
 		const user = await this.userModel
 			.query()
-			.select("id", "firstName", "lastName", "passwordHash", "passwordSalt")
-			.where("email", email)
+			.select(
+				"users.id",
+				"users.first_name",
+				"users.last_name",
+				"users.password_hash as passwordHash",
+				"users.password_salt as passwordSalt",
+				"users.group_id as groupId",
+			)
+			.withGraphJoined("group.permissions")
+			.where("users.email", email)
 			.first();
 
-		return user ?? null;
+		if (!user || !user.group || !user.group.permissions) {
```

Comment:
we only need to check if user is present, the group is always present if you joined it and permissions should be an array by default. If it's possible for them to be null, it should be handled in GroupEntity.initialize 
```suggestion
		if (!user) {
```